query-creator/create-query/tests/unit/test-handler.js

```diff
@@ -115,280 +139,442 @@ describe('Denormalize resource', function () {
+          null AS res_2_name,
+          null AS res_2_learner_id,
+          null AS res_2_remote_endpoint,
+          null AS res_2_resource_url,
+          null AS res_2_last_run,
+          null AS res_2_total_num_questions,
+          null AS res_2_total_num_answers,
+          null AS res_2_total_percent_complete,
+          null AS res_2_num_required_questions,
+          null AS res_2_num_required_answers,
+          activities_1.questions['multiple_choice_00000'].prompt AS res_1_multiple_choice_00000_choice,
+          activities_1.questions['multiple_choice_01000'].prompt AS res_1_multiple_choice_01000_choice,
+          activities_1.questions['multiple_choice_02000'].prompt AS res_1_multiple_choice_02000_choice,
+          activities_1.questions['multiple_choice_03000'].prompt AS res_1_multiple_choice_03000_choice,
+          null AS res_1_multiple_choice_03000_submitted,
+          activities_1.questions['open_response_11111'].prompt AS res_1_open_response_11111_text,
+          activities_1.questions['open_response_22222'].prompt AS res_1_open_response_22222_text,
+          null AS res_1_open_response_22222_submitted,
+          activities_1.questions['image_question_33333'].prompt AS res_1_image_question_33333_image_url,
+          activities_1.questions['image_question_33333'].prompt AS res_1_image_question_33333_text,
+          activities_1.questions['image_question_33333'].prompt AS res_1_image_question_33333_answer,
+          activities_1.questions['image_question_44444'].prompt AS res_1_image_question_44444_image_url,
+          activities_1.questions['image_question_44444'].prompt AS res_1_image_question_44444_text,
+          activities_1.questions['image_question_44444'].prompt AS res_1_image_question_44444_answer,
+          null AS res_1_image_question_44444_submitted,
+          activities_1.questions['managed_interactive_55555'].prompt AS res_1_managed_interactive_55555_text,
+          activities_1.questions['managed_interactive_66666'].prompt AS res_1_managed_interactive_66666_choice,
+          activities_1.questions['managed_interactive_77777'].prompt AS res_1_managed_interactive_77777_image_url,
+          activities_1.questions['managed_interactive_77777'].prompt AS res_1_managed_interactive_77777_text,
+          activities_1.questions['managed_interactive_77777'].prompt AS res_1_managed_interactive_77777_answer,
+          activities_1.questions['managed_interactive_88888'].prompt AS res_1_managed_interactive_88888_json,
+          activities_1.questions['managed_interactive_88888'].prompt AS res_1_managed_interactive_88888_url,
+          activities_1.questions['managed_interactive_99999'].prompt AS res_1_managed_interactive_99999_json,
+          activities_2.questions['managed_interactive_88888'].prompt AS res_2_managed_interactive_88888_json,
+          activities_2.questions['managed_interactive_88888'].prompt AS res_2_managed_interactive_88888_url
+        FROM activities_1, activities_2
```

Comment:
I think this from will have the same problems. When activities_1 is empty then the result of this query will be empty.

Reply:
Fixed in https://github.com/concord-consortium/report-service/pull/178/commits/036a56584b621c5049c13ea4d0e47a031c5f91bd